project/TestsPlugin.scala

```diff
@@ -51,24 +51,14 @@ object TestsPlugin extends AutoPlugin {
     scalaVersion := testScalaVersion.value,
   )
 
-  private val runIntegrationTest = Def.task {
-    val confFile = baseDirectory.value / "test.conf"
+  private val runIntegrationTest = (baseDirectory, dependencyResolution, streams).map { (baseDir, depRes, streams) =>
+    val confFile = baseDir / "test.conf"
     val conf = ConfigFactory.parseFile(confFile).resolve()
     val moduleBase = conf.getString("groupId") % conf.getString("artifactId")
-    val depRes = dependencyResolution.value
-    val oldJar = getArtifact(depRes, moduleBase % conf.getString("v1"), streams.value.log)
-    val newJar = getArtifact(depRes, moduleBase % conf.getString("v2"), streams.value.log)
-    streams.value.log.info(s"Comparing $oldJar -> $newJar")
-    runCollectProblemsTest(
-      (functionalTests / Compile / fullClasspath).value, // the test classpath from the functionalTest project for the test
-      (functionalTests / scalaInstance).value, // get a reference to the already loaded Scala classes so we get the advantage of a warm jvm
-      streams.value,
-      thisProjectRef.value.project,
-      oldJar,
-      newJar,
-      baseDirectory.value,
-      scalaVersion.value,
-      confFile.getAbsolutePath)
+    val oldJar = getArtifact(depRes, moduleBase % conf.getString("v1"), streams.log)
+    val newJar = getArtifact(depRes, moduleBase % conf.getString("v2"), streams.log)
+    streams.log.info(s"Comparing $oldJar -> $newJar")
+    runCollectProblemsTest(oldJar, newJar, confFile.getAbsolutePath)
```

Comment:
Boringly, this is where the bug is.

In order to dedup, I moved the shared core (including the `.value` invocations) into `runCollectProblemsTest`, redefining it with a `Def.task` wrapper.  Which means it now returns an `Initialise[Task[Unit]]]`, not a `Unit`.  That means, in turn, that `runIntegrationTest`'s type is now `Initialise[Task[Initialise[Task[Unit]]]`.

But of course `IntegrationTest / test := runIntegrationTest.value` doesn't care!  It's happy to retrieve that internal `Initialise[Task[Unit]]`, and then **discard it** and insert unit... 🙄